docs/site/migration/models/mixins.md

```diff
@@ -6,8 +6,805 @@ sidebar: lb4_sidebar
+
+In LoopBack 3, a developer is able to create a model property mixin by:
+
+- placing the mixin logic in a file in a mixins directory
+- updating the server/model-config.json file with the mixin directory location
+- updating the model's json file to include the mixin's name and a boolean
+
+As an example, we will create a mixin that adds a **category** property to a
+model.
+
+### Defining The Model Property Mixin category.js
+
+The developer defines a model property mixin in **common/mixins/category.js**
+which adds a required property named `category` to any model.
+
+{% include code-caption.html content="common/mixins/category.js" %}
+
+```js
+module.exports = function(Model, options) {
+  Model.defineProperty('category', {type: 'string', required: true});
+};
+```
+
+### Updating model-config.json
+
+The **server/model-config.json** needs to contain:
+
+- the locations of all **models**
+- the location of all **mixins**
+- the entry of the model that receives the mixin content (for this example
+  `Note`)
+
+{% include code-caption.html content="server/model-config.json" %}
+
+```
+{
+  "_meta": {
+    "sources": [
+      "loopback/common/models",
+      "loopback/server/models",
+      "../common/models",
+      "./models"
+    ],
+    "mixins": [
+      "loopback/common/mixins",
+      "loopback/server/mixins",
+      "../common/mixins",
+      "./mixins"
+    ]
+  },
+
+ // ... other entries
+
+  "Note": {
+    "dataSource": "db"
+  }
+}
+
+```
+
+### Applying The category.js Mixin To A Model
+
+To extend the model `Note` with the **category.js** mixin, we need to add a
+**mixins** section in **common/models/note.json** to indicate which mixins
+should be applied to it.
+
+{% include code-caption.html content="common/models/note.json" %}
+
+```json
+{
+  "name": "Note",
+  "properties": {
+    "title": {
+      "type": "string",
+      "required": true
+    },
+    "content": {
+      "type": "string"
+    }
+  },
+  "mixins": {
+    "Category": true
+  }
+}
+```
+
+Specifying a value of **true** for `Category` will apply the **category.js**
+property model mixin to the `Note` model. A value of **false** will not apply
+the mixin.
+
+## LoopBack 4 Approach To Creating A Model Property Mixin
+
+In LoopBack 4, a developer is able to create a model property mixin by:
+
+- placing the mixin class factory function in a separate file
+- generating a model using the CLI as usual
+- adjusting the model file to make use of the mixin class factory function
+
+### Defining The Model Property Mixin Class Factory Function
+
+This mixin class factory function `addCategoryPropertyMixin` in
+**src/mixins/categoryPropertyMixin.ts** adds the required property **category**
+to any model.
+
+{% include code-caption.html content="src/mixins/categoryPropertyMixin.ts" %}
+
+```ts
+import {Constructor} from '@loopback/context';
+
+import {property} from '@loopback/repository';
+
+export function addCategoryPropertyMixin<T extends Constructor<any>>(
```

Comment:
We usually use PascalCase naming for mixins applied on class constructors, see e.g. https://github.com/strongloop/loopback-next/blob/28ebb8b7276577494d398423104d124b80241d43/examples/todo/src/application.ts#L15-L17

```suggestion
export function AddCategoryPropertyMixin<T extends Constructor<any>>(
```

Reply:
@bajtos , I did look at how Mixins were used in lb4, and at the moment, they only center around the main application. :)